### DISCUSSION TITLE
Errors in NWB file type while running nwbinspector script

### DISCUSSION THREAD

**Discussion by Icardij** [dandi/helpdesk#71] [Q&A]
Hi,

I'm relatively new to working with nwb files and Anaconda, and am currently performing a test upload to the DANDI archive staging site, however I have error that keeps showing up that is preventing me from moving beyond the nwbinspector step in the hand book (step 2.b.ii at dandiarchive.org/handbook/10_using_dandi/#uploading-a-dandiset) I've attached a image of the complete print out from running the command below. 

![image](https://user-images.githubusercontent.com/93269545/164934291-be76b2e3-0a16-4675-9af3-cac64bcc744b.png)

The file "test.nwb" was created using the matNWB format, and reads as a NWB file type in file explorer. I am also having a issue where the command for keyring that allows you to store my API key doesn't accept a input, however that seems like it should effect the above script. If someone knows what the issue is and how to fix it that would be great. 

Thanks

**Comment by bendichter** [comment]
My guess is you aren't giving the correct relative path for the NWB file. Is the NWB file in the directory where you are running that command? Try cd ing into that directory (let me know if that makes sense)

**Reply by Icardij** [reply]
I tried to cd directly to the file my NWB file is in, and when I run the command it gives back a new error. 
"UnboundLocalError: local variable 'nwbfile' referenced before assignment"

**Comment by bendichter** [comment]
@codycbakerphd can you help with this?

**Comment by CodyCBakerPhD** [comment]
@Icardij Hi there, sorry to hear you're having trouble with the NWBInspector - would it be possible for you to share the NWBFile in question? If possible, email to cody.baker@catalystneuro.com and I can take a deep dive into what might be going wrong

**Comment by CodyCBakerPhD** [comment]
Turns out that while the NWBInspector failed to catch the error and include it as a message in the report (hotfix for that found at https://github.com/NeurodataWithoutBorders/nwbinspector/pull/204, will issue bug fix release shortly after), the file was not loadable in PyNWB. Issues specific to their files will be relegated to either MatNWB or PyNWB depending on whose side of the API the issue is with.
